service/kvmclusterconfig/v1/key/key_test.go

```diff
@@ -0,0 +1,177 @@
+		},
+		{
+			CustomObject: v1alpha1.KVMClusterConfig{
+				Spec: v1alpha1.KVMClusterConfigSpec{
+					Guest: v1alpha1.KVMClusterConfigSpecGuest{
+						ClusterGuestConfig: v1alpha1.ClusterGuestConfig{
+							ID:   "cluster-123",
+							Name: "First cluster",
+						},
+					},
+				},
+			},
+			ExpectedSecretName: "cluster-123-encryption",
+		},
+	}
+
+	for i, tc := range testCases {
+		encryptionKeySecretName := EncryptionKeySecretName(tc.CustomObject)
+		if encryptionKeySecretName != tc.ExpectedSecretName {
+			t.Errorf("TestCase %d: EncryptionKeySecretName %s doesn't match. Expected: %s",
+				(i + 1), encryptionKeySecretName, tc.ExpectedSecretName)
+		}
+	}
+
+}
+
+func Test_ToCustomObject(t *testing.T) {
+	var emptyKVMClusterConfigPtr *v1alpha1.KVMClusterConfig
+
+	testCases := []struct {
+		InputObject          interface{}
+		ExpectedCustomObject v1alpha1.KVMClusterConfig
+		ExpectedError        error
+	}{
+		// Success case - pass empty KVMClusterConfig reference
+		{
+			InputObject:          &v1alpha1.KVMClusterConfig{},
+			ExpectedCustomObject: v1alpha1.KVMClusterConfig{},
+			ExpectedError:        nil,
+		},
+		// Success case - verify that internal fields are transferred as well
+		{
+			InputObject: &v1alpha1.KVMClusterConfig{
+				Spec: v1alpha1.KVMClusterConfigSpec{
+					Guest: v1alpha1.KVMClusterConfigSpecGuest{
+						ClusterGuestConfig: v1alpha1.ClusterGuestConfig{
+							ID:   "cluster-1",
+							Name: "My own snowflake cluster",
+						},
+					},
+				},
+			},
+			ExpectedCustomObject: v1alpha1.KVMClusterConfig{
+				Spec: v1alpha1.KVMClusterConfigSpec{
+					Guest: v1alpha1.KVMClusterConfigSpecGuest{
+						ClusterGuestConfig: v1alpha1.ClusterGuestConfig{
+							ID:   "cluster-1",
+							Name: "My own snowflake cluster",
+						},
+					},
+				},
+			},
+			ExpectedError: nil,
+		},
+		// Failure : Nil *v1alpha1.KVMClusterConfig is not accepted.
+		{
+			InputObject:          emptyKVMClusterConfigPtr,
+			ExpectedCustomObject: v1alpha1.KVMClusterConfig{},
+			ExpectedError:        emptyValueError,
+		},
+		// Failure: Non-reference value of KVMClusterConfig is not accepted.
+		{
+			InputObject:          v1alpha1.KVMClusterConfig{},
+			ExpectedCustomObject: v1alpha1.KVMClusterConfig{},
+			ExpectedError:        wrongTypeError,
+		},
+		// Failure: Wrong type must not be accepted
+		{
+			InputObject:          &v1alpha1.AzureClusterConfig{},
+			ExpectedCustomObject: v1alpha1.KVMClusterConfig{},
+			ExpectedError:        wrongTypeError,
+		},
+		// Failure: Nil interface{} must fail
+		{
+			InputObject:          nil,
+			ExpectedCustomObject: v1alpha1.KVMClusterConfig{},
+			ExpectedError:        wrongTypeError,
+		},
+	}
+
+	for i, tc := range testCases {
+		customObject, err := ToCustomObject(tc.InputObject)
+		if microerror.Cause(err) != tc.ExpectedError {
+			t.Errorf("TestCase %d: Received error %#v doesn't match expected %#v",
+				(i + 1), err, tc.ExpectedError)
```

Comment:
- You actually do not need the braces here in order to do the calculation.
- In general we want to index test cases at 0, so not need to add 1.

Reply:
I personally find that braces increase readability, but I'm also fine without them, though with the second point they are not needed. I'll fix these after getting open PRs merged so that I can fix them from everywhere at one go.